src/dune_rules/pkg_rules.ml

```diff
@@ -1415,8 +1455,18 @@ module Gen_rules = Build_config.Gen_rules
 
 let setup_package_rules context ~dir ~pkg_name : Gen_rules.result Memo.t =
   let name = User_error.ok_exn (Package.Name.of_string_user_error (Loc.none, pkg_name)) in
-  let* db = Lock_dir.get context in
-  let+ pkg = resolve db.packages context (Loc.none, name) in
+  let* db = DB.get context in
+  let+ pkg =
+    Resolve.resolve db context (Loc.none, name)
+    >>| function
+    | `Inside_lock_dir pkg -> pkg
+    | `System_provided ->
+      User_error.raise
+        [ Pp.textf
+            "no rules for %s because it's set as provided by the system"
```

Comment:
Can you change this error to "There are no rules for ..." for clarity (assuming I've interpreted it correctly)